pallets/nodle-staking/src/lib.rs

```diff
@@ -0,0 +1,1566 @@
+            ensure!(!state.is_leaving(), Error::<T>::AlreadyLeaving);
+            let mut exits = <ExitQueue<T>>::get();
+            let now = <Round<T>>::get().current;
+            let when = now + T::BondDuration::get();
+            ensure!(
+                exits.insert(Bond {
+                    owner: validator.clone(),
+                    amount: when,
+                }),
+                Error::<T>::AlreadyLeaving,
+            );
+            state.leave_validators_pool(when);
+            let mut candidates = <ValidatorPool<T>>::get();
+            if candidates.remove(&Bond::from_owner(validator.clone())) {
+                <ValidatorPool<T>>::put(candidates);
+            }
+            <ExitQueue<T>>::put(exits);
+            <ValidatorState<T>>::insert(&validator, state);
+            Self::deposit_event(Event::ValidatorScheduledExit(now, validator, when));
+            Ok(().into())
+        }
+        /// Rejoin the set of validators pool
+        /// if previously had called `go_validator_offline`
+        #[pallet::weight(10_000)]
+        pub fn go_validator_online(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+            let validator = ensure_signed(origin)?;
+            let mut state = <ValidatorState<T>>::get(&validator).ok_or(Error::<T>::ValidatorDNE)?;
+            ensure!(!state.is_active(), Error::<T>::AlreadyActive);
+            ensure!(!state.is_leaving(), Error::<T>::CannotActivateIfLeaving);
+            state.go_online();
+            let mut validators = <ValidatorPool<T>>::get();
+            ensure!(
+                validators.insert(Bond {
+                    owner: validator.clone(),
+                    amount: state.total
+                }),
+                Error::<T>::AlreadyActive
+            );
+            <ValidatorPool<T>>::put(validators);
+            <ValidatorState<T>>::insert(&validator, state);
+            Self::deposit_event(Event::ValidatorBackOnline(
+                <Round<T>>::get().current,
+                validator,
+            ));
+            Ok(().into())
+        }
+        /// Temporarily leave validators pool without unbonding
+        #[pallet::weight(10_000)]
+        pub fn go_validator_offline(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+            let validator = ensure_signed(origin)?;
+            let mut state = <ValidatorState<T>>::get(&validator).ok_or(Error::<T>::ValidatorDNE)?;
+            ensure!(state.is_active(), Error::<T>::AlreadyOffline);
+            state.go_offline();
+            let mut validators = <ValidatorPool<T>>::get();
+            if validators.remove(&Bond::from_owner(validator.clone())) {
+                <ValidatorPool<T>>::put(validators);
+            }
+            <ValidatorState<T>>::insert(&validator, state);
+            Self::deposit_event(Event::ValidatorWentOffline(
+                <Round<T>>::get().current,
+                validator,
+            ));
+            Ok(().into())
+        }
```

Comment:
I'd say we do not need this for the sake of simplicity

Reply:
Sure, removed it :+1: